src/Cli/dotnet/commands/dotnet-workload/install/NetSdkManagedInstaller.cs

```diff
@@ -230,6 +242,8 @@ private void DeletePack(PackInfo packInfo)
         {
             if (PackIsInstalled(packInfo))
             {
+                GC.Collect();
+                GC.WaitForPendingFinalizers();
```

Comment:
What's this for?  Do we have any reason to think this will make something more reliable?

Reply:
Yes, this change is what actually fixes #17206, I've confirmed that it prevents us from trying to delete a locked file.